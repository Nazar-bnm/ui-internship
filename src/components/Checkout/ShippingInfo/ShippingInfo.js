import React from 'react';
import cx from 'classnames';

const CN = 'shipping-info';

class ShippingInfo extends React.Component {
  render() {
    return (
      <div className={cx(CN)}>
       ShippingInfo
      </div>
    );
  }
}

export default ShippingInfo;
