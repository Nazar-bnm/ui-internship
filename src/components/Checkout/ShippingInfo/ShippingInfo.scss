@import '../../../scss/global_styles';

.shipping-info {
  padding: 20px;
  font-family: font(primary-light);
  text-align: justify;
}