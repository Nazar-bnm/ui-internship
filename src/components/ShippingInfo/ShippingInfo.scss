@import '../../scss/global_styles';

.shippingInfo {
  justify-content: center;
  
  &__list {
    display: flex;
    justify-content: space-between;
    list-style: none;
    margin: 30px 0;
    width: 80%;

    &-item {
      display: flex;
      width: 30%;
    }
  }

  &__item {
    &-number {
      font-family: font(secondary);
      font-size: 1.4em;
    }

    &-text {
      &__wrapper {
        text-transform: uppercase;
        margin-left: 10px;
      }

      &__title {
        font-weight: 300;
        margin-bottom: 3px;
      }

      &__description {
        font-family: font(primary);
        font-size: 0.6em;
      }
    }
  }

  &__line {
    width: 100%;
    border: 0.5px solid $alto;
  }
}

@media screen and (max-width: 768px) {
  .shippingInfo__list {
    width: 100%;
  }
}

@media screen and (max-width: 480px) {
  .shippingInfo__list {
    &-item {
      flex-direction: column;
      text-align: center;
    }
  }

  .shippingInfo__item-text {
    &__wrapper {
      margin-left: 0;
      padding-top: 7px;
    }

    &__title {
      font-size: 0.8em;
    }
  }
}
