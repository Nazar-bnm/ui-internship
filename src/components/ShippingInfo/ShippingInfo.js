import React from 'react';

import cx from 'classnames';

import './ShippingInfo.scss';

export const CN = 'shippingInfo';

const ShippingInfo = () => {
  const shippingData = [
    { number: 1, title: 'free shipping', description: 'on all orders over 50$' },
    { number: 2, title: 'money back guarantee', description: 'on all orders' },
    { number: 3, title: 'worldwide delivery', description: 'to over 80 countries' },
  ];
  const shippingInfo = shippingData.map((elem) => (
    <li key={elem.number} className={`${CN}__list-item`}>
      <span className={`${CN}__item-number`}>0{elem.number}.</span>
      <div className={`${CN}__item-text__wrapper`}>
        <h5 className={`${CN}__item-text__title`}>{elem.title}</h5>
        <h6 className={`${CN}__item-text__description`}>{elem.description}</h6>
      </div>
    </li>
  ));

  return (
    <div className='content container'>
      <hr className={`${cx(CN)}__line`}/>
      <ul className={`${cx(CN)}__list`}>
        {shippingInfo}
      </ul>
    </div>
  );
};

export default ShippingInfo;
