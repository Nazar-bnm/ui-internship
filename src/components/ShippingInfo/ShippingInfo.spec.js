import React from 'react';
import { shallow } from 'enzyme';
import ShippingInfo from './ShippingInfo';

describe('<ShippingInfo />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ShippingInfo />
    );
  });

  test('renders', () => {
    wrapper = shallow(<ShippingInfo />);
    expect(wrapper.exists()).toBe(true);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
