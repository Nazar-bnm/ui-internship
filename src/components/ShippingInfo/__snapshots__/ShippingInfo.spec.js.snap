// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingInfo /> should match snapshot 1`] = `
<div
  className="content container"
>
  <hr
    className="shippingInfo__line"
  />
  <ul
    className="shippingInfo__list"
  >
    <li
      className="shippingInfo__list-item"
      key="1"
    >
      <span
        className="shippingInfo__item-number"
      >
        0
        1
        .
      </span>
      <div
        className="shippingInfo__item-text__wrapper"
      >
        <h5
          className="shippingInfo__item-text__title"
        >
          free shipping
        </h5>
        <h6
          className="shippingInfo__item-text__description"
        >
          on all orders over 50$
        </h6>
      </div>
    </li>
    <li
      className="shippingInfo__list-item"
      key="2"
    >
      <span
        className="shippingInfo__item-number"
      >
        0
        2
        .
      </span>
      <div
        className="shippingInfo__item-text__wrapper"
      >
        <h5
          className="shippingInfo__item-text__title"
        >
          money back guarantee
        </h5>
        <h6
          className="shippingInfo__item-text__description"
        >
          on all orders
        </h6>
      </div>
    </li>
    <li
      className="shippingInfo__list-item"
      key="3"
    >
      <span
        className="shippingInfo__item-number"
      >
        0
        3
        .
      </span>
      <div
        className="shippingInfo__item-text__wrapper"
      >
        <h5
          className="shippingInfo__item-text__title"
        >
          worldwide delivery
        </h5>
        <h6
          className="shippingInfo__item-text__description"
        >
          to over 80 countries
        </h6>
      </div>
    </li>
  </ul>
</div>
`;
